src/roswire/app/app.py

```diff
@@ -155,3 +155,39 @@ def launch(
             app=self, dockerblade=container, host_workspace=host_workspace
         )
         return instance
+
+    def attach(
+        self,
+        container_name_or_id: str,
+        *,
+        require_description: bool = True
+    ) -> AppInstance:
+        """Attaches to a running container for this application.
+
+        Parameters
+        ----------
+        container_name_or_id: str
+            The name or ID of the container to which ROSWire should be
+            attached.
+        require_description: bool
+            If :code:`True`, this method call will ensure that there is a
+            description of the application before launching. If :code:`False`,
+            no check will be performed.
+
+        Returns
+        -------
+        AppInstance
+            A ROSWire interface around the given container.
+        """
+        dockerblade = self._roswire._dockerblade
```

Comment:
Is is proper to access a protected member of a `_roswire` in this way?

Reply:
I think this is fine in this particular case. We don't necessarily want to expose `_dockerblade` to external clients of ROSWire, but internally that information is useful. This roughly fits the intuition behind Java's `protected` modifier.